Sources/JXKit/JXValue.swift

```diff
@@ -454,14 +503,72 @@ extension JXValue {
         }
     }
 
-    /// Returns the JavaScript object as dictionary.
-    @inlinable public var dictionary: [String: JXValue]? {
+    /// Returns the JavaScript object as a dictionary.
+    @inlinable public var dictionary: [String: JXValue] {
         get throws {
-            !isObject ? nil : try self.properties.reduce(into: [:]) { $0[$1] = try self[$1] }
+            try self.properties.reduce(into: [:]) { result, property in
+                result[property] = try self[property]
+            }
         }
     }
 }
 
+// When conveying to a specified type, we use function overloads to access the desired optional, array, or dictionary element type.
```

Comment:
This bit is tricky. We want to be able to pull out the optional/array/dict element type from the given "to" type. So we create function overloads for Optional, Array, Dictionary, Optional Array, Optional Dictionary. And then we can decide how to handle individual elements, including letting the SPI have a crack and conveying them.